src/WebServer/Http/HttpServer.cs

```diff
@@ -66,14 +68,28 @@ public void StopServer()
                     using (var inputStream = args.Socket.InputStream)
                     {
                         var request = await MutableHttpServerRequest.Parse(inputStream);
+                        if (!request.IsComplete)
+                        {
+                            _log.Warn($"Request {request.Method} {request.Uri} aborted.");
+                            return;
+                        }
+
+                        if (_log.IsDebugEnabled())

```

Comment:
Should this check be moved to the `Debug` method itself. Pretty error prone if you have to check this yourself everytime.